atlasdb-dbkvs-hikari/src/main/java/com/palantir/nexus/db/pool/HikariCPConnectionManager.java

```diff
@@ -84,18 +85,34 @@ public HikariCPConnectionManager(ConnectionConfig connConfig) {
 
     @Override
     public Connection getConnection() throws SQLException {
-        long start = System.currentTimeMillis();
-        try {
-            return getConnectionInternal();
-        } finally {
-            long now = System.currentTimeMillis();
-            long elapsed = now - start;
-            if (elapsed > 1000) {
-                log.warn("Waited {}ms for connection", elapsed);
-                logPoolStats();
-            } else {
-                log.debug("Waited {}ms for connection", elapsed);
+        while (true) {
+            long start = System.currentTimeMillis();
+            Pair<HikariDataSource, Connection> ret;
+            try {
+                ret = getConnectionInternal();
```

Comment:
rather than using a pair, getConnectionInternal() could become getDataSourcePoolInternal() and then call getConnection() explicitly here (I think this is a little bit cleaner than making and unpacking a Pair)